pkg/skaffold/docker/default_repo.go

```diff
@@ -19,16 +19,25 @@ package docker
 import (
 	"regexp"
 	"strings"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker/registry"
 )
 
 const maxLength = 255
 
 var (
-	escapeRegex = regexp.MustCompile(`[/._:@]`)
 	prefixRegex = regexp.MustCompile(`(.*\.)?gcr.io/[a-zA-Z0-9-_]+/?`)
+	escapeRegex = regexp.MustCompile(`[/._:@]`)
 )
 
-func SubstituteDefaultRepoIntoImage(defaultRepo string, image string) (string, error) {
+func SubstituteDefaultRepoIntoImage(defaultRepo string, image string, rewriteStrategy bool) (string, error) {
```

Comment:
TBH I found this code hard to parse and reason through.  I had to scroll back and forth to track what was going on.

I wonder if it could be simplified by instead representing the strategies as objects, and the appropriate strategy is chosen based on the defaultRepo and rewriteStrategy.

Reply:
I refactored some more to keep the changes minimal. Please take a look and we can sync in person tomorrow?